app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -510,19 +508,14 @@ class LuciBuildService {
     return false;
   }
 
-  bool _shouldRerunBuilder(LuciTask luciTask, String repo) {
-    if (luciTask.summaryMarkdown == null || !luciTask.builderName.contains('Mac')) {
+  /// Only blocking failures on framework TOT will be triggered.
+  Future<bool> _shouldRerunBuilder(LuciTask luciTask, int retries, Commit commit, DatastoreService datastore) async {
+    if ((luciTask.status != Task.statusFailed && luciTask.status != Task.statusInfraFailure) ||
```

Comment:
Creating a set of the expected status and comparing the current one with the 'in' operator will make this logic simpler.

Reply:
Done.